src/pivot-table/hooks/use-column-width.ts

```diff
@@ -282,11 +287,16 @@ export default function useColumnWidth(
    * Gets the width of a right grid column. This is always based on the leaf width(s)
    */
   const getRightGridColumnWidth = useCallback(
-    (index?: number) =>
-      topGridLeavesIsPseudo && index !== undefined
-        ? leafWidths[layoutService.getMeasureInfoIndexFromCellIndex(index)]
-        : averageLeafWidth,
-    [topGridLeavesIsPseudo, leafWidths, layoutService, averageLeafWidth],
+    (index?: number) => {
+      const scrollbarWidthSharePerColumn = !allRowsVisible
+        ? parseFloat((verticalScrollbarWidth / layoutService.size.x).toFixed(12))
+        : 0;
+
+      return topGridLeavesIsPseudo && index !== undefined
+        ? leafWidths[layoutService.getMeasureInfoIndexFromCellIndex(index)] - scrollbarWidthSharePerColumn
+        : averageLeafWidth - scrollbarWidthSharePerColumn;
```

Comment:
How come we remove the scrollbar width at this point? Can't you do it when you calculate leafWidths? By just subtracting the width from `rightGridAvailableWidth`?

Reply:
because getRightGridColumnWidth() is the unified way we calculate columns width one by one -> by passing it to the VT related components. 

The point is we dont have a single value indicating the entire width of **all** columns at once, (we may calculate it at some point but it's not been used in VT components to define the width), every single column width being calculated one by one and that is why I chosed to go this way.

Plus, `rightGridAvailableWidth` is the viewable area in right grid, not the (full) width we use for columns, which is the final guy who decides on (sum) width of all columns

Reply:
So i gave my idea a try. Didn't get the calculations correct, so I think we can keep the `- scrollbarWidthSharePerColumn` here